contracts/admin/src/common.rs

```diff
@@ -0,0 +1,20 @@
+use serde::{Deserialize, Serialize};
+
+use sdk::schemars::{self, JsonSchema};
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct GeneralContractsGroup<T> {
+    pub profit: T,
+    pub timealarms: T,
+    pub treasury: T,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct SpecializedContractsGroup<T> {
```

Comment:
probably I do not get the idea, but why the dispatcher and leaser are part of it?

Reply:
I have stored all contracts directly based off a specific currency and the ones which refer to such instances into this group. Dispatcher holds reference to LPP and Oracle, which both as far as I understand have one currency and there would be an instance for each base. Same story with Leaser.

I might have misunderstood you when discussed offline.

Reply:
SpecializedContractsGroup -> LpnContracts